sklearn/calibration.py

```diff
@@ -408,6 +447,14 @@ def fit(self, X, y, sample_weight=None):
             self.label_encoder_.fit(self.classes)
 
         self.classes_ = self.label_encoder_.classes_
+        X = np.array(X) if isinstance(X, list) else X
+        sample_weight = _check_sample_weight(sample_weight, X,
+                                             dtype=X.dtype)
+        self.class_weight_ = compute_class_weight(self.class_weight,
+                                                  self.classes_, y)
+        sample_weight *= self.class_weight_[
+            _unique(column_or_1d(y), return_inverse=True)[1]]
+
```

Comment:
Hi,
Thank you very much for your comments, I am looking into them and will push an update version later today.
I am not using the `fit_transform` from `LabelEncoder` to avoid creating another instance of the class but I think I will follow your example for clarity.